### PR TITLE
Remove the 'What is the developer preview' from docs

### LINKED ISSUE #2873: Delete "What is the Raiden Developer Preview"
## Abstract

Delete "What is the Raiden Developer Preview" within the documentation (see https://raiden-network.readthedocs.io/en/latest/what_is_the_dev_preview.html)

## Motivation

It is outdated due to the testnet releases

### PR DESCRIPTION
**Pull request by palango**
Fixes #2873

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -21,7 +21,6 @@ Contents
   overview_and_guide
   api_walkthrough
   rest_api
-  what_is_the_dev_preview
```
Comment (czepluch) [inline docs/index.rst]: I'd actually keep the doc itself in the repo, but just not link to it. Just for historical reasons. But I don't have a strong opinion.

Comment (palango) [reply docs/index.rst]: Fine for me, but we can always restore it from git if necessary.

**Comment by LefterisJP** [comment]
Shouldn't we also remove the actual rst file?

**Comment by palango** [comment]
See https://github.com/raiden-network/raiden/pull/2876#discussion_r227717965

File: docs/index.rst (hunk 1 above)
Comment (LefterisJP) [reply docs/index.rst]: Let's remove it. It's in git.

**Pull request merged**